src/creeps/Zombie.h

```diff
@@ -1,33 +1,183 @@
 #ifndef ZOMBIE_H
 #define ZOMBIE_H
 #include <string>
+#include <math.h>
+#include <random>
+#include <vector>
+#include <utility>
+#include <SDL2/SDL.h>
 #include "../sprites/LTexture.h"
 #include "../collision/HitBox.h"
 #include "../basic/Entity.h"
 #include "../collision/CollisionHandler.h"
 #include "../inventory/Inventory.h"
-#include <math.h>
-#include <vector>
-#include <SDL2/SDL.h>
+#include "../buildings/Base.h"
 #include "../view/Window.h"
 #include "../basic/Movable.h"
 
-constexpr static int ZOMBIE_HEIGHT = 125;
-constexpr static int ZOMBIE_WIDTH = 75;
+static constexpr int ZOMBIE_VELOCITY = 50;
+static constexpr int ZOMBIE_INIT_HP  = 100;
+static constexpr int ZOMBIE_FRAMES   = 50;
+static constexpr int ZOMBIE_HEIGHT   = 125;
+static constexpr int ZOMBIE_WIDTH    = 75;
+
+// 8 possible directions
+static constexpr int DIR_CAP = 8;
+enum ZombieDirections {
+    DIR_R,
+    DIR_RD,
+    DIR_D,
+    DIR_LD,
+    DIR_L,
+    DIR_LU,
+    DIR_U,
+    DIR_RU
+};
+
+// Cardinal directions for setting angles
+enum ZombieAngles {
+    NORTH = 0,
+    NORTHEAST = 45,
+    EAST = 90,
+    SOUTHEAST = 135,
+    SOUTH = 180,
+    SOUTHWEST = 225,
+    WEST = 270,
+    NORTHWEST = 315
+};
+
+// overlapped
+static constexpr float OVERLAP = 0.1f;
```

Comment:
'f' is not needed when variable is defined as float
Either declare it as auto, or remove the f